### PR TITLE
Feat(eos_cli_config_gen): Added dot1x radius av-pair `lldp` and `dhcp` command support

### LINKED ISSUE #4577: Support for additional dot1x commands
### Enhancement summary

We are working on a project that requires Cisco ISE Device Profiling with LLDP TLVs and DHCP options. Device profiling feature allows EOS to send attributes of the authenticated hosts to the Accounting server in RADIUS Accounting messages.

In order to support that we need to enable the following commands in dot1x config mode:

```
Device profiling with DHCP options can be enabled by the following command under config-dot1x mode:
[ no | default ] radius av-pair dhcp ( hostname | parameter-request-list | vendor-class-id ) [ auth-only ]

Similarly, device profiling with LLDP TLVs can be enabled by the following command under config-dot1x mode:
[ no | default ] radius av-pair lldp ( system-name | system-description ) [ auth-only ]
```

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Cisco ISE endpoint profiling, for example:
```
dot1x
   radius av-pair lldp system-name auth-only
   radius av-pair lldp system-description
   radius av-pair dhcp hostname
   radius av-pair dhcp parameter-request-list
   radius av-pair dhcp vendor-class-id
```

### Describe the solution you would like

Support for the previous commands via eos_cli_config_gen

### Describe alternatives you have considered

Not found any today

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Added dot1x radius av-pair `lldp` and `dhcp` command support

## Related Issue(s)

Fixes #4577 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Added dot1x radius av-pair `lldp` and `dhcp` command support

## How to test
Run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 1]
```diff
@@ -108,6 +108,43 @@ dot1x
    captive-portal start limit infinite
 {%             endif %}
 {%         endif %}
+{%         if dot1x.radius_av_pair.lldp is arista.avd.defined or dot1x.radius_av_pair.dhcp is arista.avd.defined %}
+{%             if dot1x.radius_av_pair.lldp.system_name.enabled is arista.avd.defined(true) %}
+{%                 set av_pair_lldp = "radius av-pair lldp system-name" %}
+{%                 if dot1x.radius_av_pair.lldp.system_name.auth_only is arista.avd.defined(true) %}
+{%                     set av_pair_lldp = "radius av-pair lldp system-name auth-only" %}
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]:
```suggestion
{%                     set av_pair_lldp = av_pair_lldp ~ " auth-only" %}
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 2]
```diff
@@ -108,6 +108,43 @@ dot1x
...
+{%                 endif %}
+   {{ av_pair_lldp }}
+{%             endif %}
+{%             if dot1x.radius_av_pair.lldp.system_description.enabled is arista.avd.defined(true) %}
+{%                 set av_pair_lldp = "radius av-pair lldp system-description" %}
+{%                 if dot1x.radius_av_pair.lldp.system_description.auth_only is arista.avd.defined(true) %}
+{%                     set av_pair_lldp = "radius av-pair lldp system-description auth-only" %}
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]:
```suggestion
{%                     set av_pair_lldp = av_pair_lldp ~ " auth-only" %}
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 3]
```diff
@@ -108,6 +108,43 @@ dot1x
...
+{%                 endif %}
+   {{ av_pair_lldp }}
+{%             endif %}
+{%             if dot1x.radius_av_pair.dhcp.hostname.enabled is arista.avd.defined(true) %}
+{%                 set av_pair_dhcp = "radius av-pair dhcp hostname" %}
+{%                 if dot1x.radius_av_pair.dhcp.hostname.auth_only is arista.avd.defined(true) %}
+{%                     set av_pair_dhcp = "radius av-pair dhcp hostname auth-only" %}
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]:
```suggestion
{%                     set av_pair_dhcp = av_pair_dhcp ~ " auth-only" %}
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 4]
```diff
@@ -108,6 +108,43 @@ dot1x
...
+{%                 endif %}
+   {{ av_pair_dhcp }}
+{%             endif %}
+{%             if dot1x.radius_av_pair.dhcp.parameter_request_list.enabled is arista.avd.defined(true) %}
+{%                 set av_pair_dhcp = "radius av-pair dhcp parameter-request-list" %}
+{%                 if dot1x.radius_av_pair.dhcp.parameter_request_list.auth_only is arista.avd.defined(true) %}
+{%                     set av_pair_dhcp = "radius av-pair dhcp parameter-request-list auth-only" %}
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]:
```suggestion
{%                     set av_pair_dhcp = av_pair_dhcp ~ " auth-only" %}
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 5]
```diff
@@ -108,6 +108,43 @@ dot1x
...
+{%                 endif %}
+   {{ av_pair_dhcp }}
+{%             endif %}
+{%             if dot1x.radius_av_pair.dhcp.vendor_class_id.enabled is arista.avd.defined(true) %}
+{%                 set av_pair_dhcp = "radius av-pair dhcp vendor-class-id" %}
+{%                 if dot1x.radius_av_pair.dhcp.vendor_class_id.auth_only is arista.avd.defined(true) %}
+{%                     set av_pair_dhcp = "radius av-pair dhcp vendor-class-id auth-only" %}
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]:
```suggestion
{%                     set av_pair_dhcp = av_pair_dhcp ~ " auth-only" %}
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 6]
```diff
@@ -108,6 +108,43 @@ dot1x
    captive-portal start limit infinite
 {%             endif %}
 {%         endif %}
+{%         if dot1x.radius_av_pair.lldp is arista.avd.defined or dot1x.radius_av_pair.dhcp is arista.avd.defined %}
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]: we can keep separate defined check for both as they are in sequence separately

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 7]
```diff
@@ -75,7 +75,7 @@ dot1x
    mac based authentication hold period {{ dot1x.mac_based_authentication.hold_period }} seconds
 {%             endif %}
 {%         endif %}
-{%         if dot1x.radius_av_pair is arista.avd.defined %}
+{%         if dot1x.radius_av_pair.service_type is arista.avd.defined(true) or dot1x.radius_av_pair.framed_mtu is arista.avd.defined %}
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]: I think this if-endif can be removed.

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 (hunk 6 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]: updated

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 (hunk 7 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]: updated

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 8]
```diff
@@ -108,6 +106,45 @@ dot1x
...
+{%                     set av_pair_lldp = av_pair_lldp ~ " auth-only" %}
+{%                 endif %}
+   {{ av_pair_lldp }}
+{%             endif %}
+{%         endif %}
+{%         if dot1x.radius_av_pair.dhcp is arista.avd.defined %}
+{%             if dot1x.radius_av_pair.dhcp.hostname.enabled is arista.avd.defined(true) %}
+{%                 set av_pair_dhcp = "radius av-pair dhcp hostname" %}
+{%                 if dot1x.radius_av_pair.dhcp.hostname.auth_only is arista.avd.defined(true) %}
+{%                     set av_pair_dhcp = av_pair_dhcp~ " auth-only" %}
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]:
```suggestion
{%                     set av_pair_dhcp = av_pair_dhcp ~ " auth-only" %}
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 9]
```diff
@@ -108,6 +106,45 @@ dot1x
...
+{%                 endif %}
+   {{ av_pair_dhcp }}
+{%             endif %}
+{%             if dot1x.radius_av_pair.dhcp.parameter_request_list.enabled is arista.avd.defined(true) %}
+{%                 set av_pair_dhcp = "radius av-pair dhcp parameter-request-list" %}
+{%                 if dot1x.radius_av_pair.dhcp.parameter_request_list.auth_only is arista.avd.defined(true) %}
+{%                     set av_pair_dhcp = av_pair_dhcp~ " auth-only" %}
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]:
```suggestion
{%                     set av_pair_dhcp = av_pair_dhcp ~ " auth-only" %}
```

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2 [hunk 10]
```diff
@@ -108,6 +106,45 @@ dot1x
...
+{%                 endif %}
+   {{ av_pair_dhcp }}
+{%             endif %}
+{%             if dot1x.radius_av_pair.dhcp.vendor_class_id.enabled is arista.avd.defined(true) %}
+{%                 set av_pair_dhcp = "radius av-pair dhcp vendor-class-id" %}
+{%                 if dot1x.radius_av_pair.dhcp.vendor_class_id.auth_only is arista.avd.defined(true) %}
+{%                     set av_pair_dhcp = av_pair_dhcp~ " auth-only" %}
```
Comment (Shivani-gslab) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/dot1x.j2]:
```suggestion
{%                     set av_pair_dhcp = av_pair_dhcp ~ " auth-only" %}
```

**Review by Shivani-gslab — CHANGES_REQUESTED** [review]

**Review by laxmikantchintakindi — APPROVED** [review]

**Review by Shivani-gslab — APPROVED** [review]
LGTM!

**Review by ClausHolbechArista — DISMISSED** [review]
Blocking until 5.0 was released

**Review by gmuloc — APPROVED** [review]

**Pull request merged**